src/v/raft/vote_stm.cc

```diff
@@ -228,6 +228,8 @@ ss::future<> vote_stm::update_vote_state(ss::semaphore_units<> u) {
     // section vote:5.2.2
     _ptr->_vstate = consensus::vote_state::leader;
     _ptr->_leader_id = _ptr->self();
+    // reset target priority
+    _ptr->_target_priority = consensus::max_voter_priority;
```

Comment:
why is max_voter_priority not a free function of naamed type. 

we should add another type to named type that is a function on `underlying_t` for named type

`named_type::max<voter_priority>()` or smth like that 

```

template<typename NT>
auto inline max() {
       return std::numeric_limits<NT::underlying_t>::max();
} 
```

or smth like that.

Reply:
could be constexpr too

Reply:
I've added `static constexpr` max/min methods to numeric `named_type`